app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx

```diff
@@ -249,9 +249,29 @@ export default function AuthProvider(props: AuthProviderProps) {
       platform: detect.platform(),
       architecture: detect.architecture(),
     })
+
     return gtagHooks.gtagOpenCloseCallback(gtagEventRef, 'open_app', 'close_app')
   }, [])
 
+  const remoteBackend = React.useMemo(() => {
+    if (session) {
+      const client = new HttpClient([['Authorization', `Bearer ${session.accessToken}`]])
+      return new RemoteBackend(client, logger, getText)
+    }
+  }, [session?.accessToken])
+
+  React.useEffect(() => {
+    if (remoteBackend) {
+      remoteBackend.logEvent('open_app')
+      const onBeforeUnload = () => remoteBackend.logEvent('close_app')
+      window.addEventListener('beforeunload', onBeforeUnload)
+      return () => {
+        window.removeEventListener('beforeunload', onBeforeUnload)
+        onBeforeUnload()
```

Comment:
`onBeforeUnload()` here looks unneeded

Reply:
it's needed, but perhaps the name is wrong - it should be `logCloseEvent` instead - the event signals that the dashboard was closed, which *is* what happens on unmount. although if `remoteBackend` ever changes in this app then it's true that it doesn't only fire on unmount...